dhcpv4/async/client.go

```diff
@@ -0,0 +1,206 @@
+package async
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/fanliao/go-promise"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// Default ports
+const (
+	DefaultServerPort = 67
+	DefaultClientPort = 68
+)
+
+// Client implements an asynchronous DHCPv4 client
+// It doesn't use the broadcast socket! Which means it should be used only when
```

Comment:
can you open an issue for the broadcast implementation, and link it here? I think it's something we may still want in the future